raiden/raiden_service.py

```diff
@@ -158,6 +163,30 @@ def send_and_wait(self, recipient, message, timeout):
 
         self.protocol.send_and_wait(recipient, message, timeout)
 
+    # api design regarding locks:
+    # - `register_secret` method was added because secret registration can be a
+    #   cross asset operation
+    # - unlocking a lock is not a cross asset operation, for this reason it's
+    #   only available in the asset manager
+
+    def register_secret(self, secret):
+        """ Register the secret with any channel that has a hashlock on it.
+
+        This must search through all channels registered for a given hashlock
+        and ignoring the assets. Useful for refund transfer, split transfer,
+        and exchanges.
+        """
+        for asset_manager in self.managers_by_asset_address.values():
+            try:
+                asset_manager.register_secret(secret)
+            except:  # pylint: disable=bare-except
+                # Only channels that care about the given secret can be
+                # registered and channels that have claimed the lock must
+                # be removed, so an excpetion should not happen at this
+                # point, nevertheless handle it because we dont want a
+                # error in a channel to mess the state from others.
+                log.error('programming error')
```

Comment:
I don't like it too much. If there is an error here, I believe we should fail and learn about the exceptions that (shall not but) may arise.